GCRCatalogs/register.py

```diff
@@ -101,17 +160,30 @@ class RootDirManager:
             GCRCatalogs.set_root_dir_by_site('sitename')
        where sitename is one of ({})
 
-       If you want to use a non-standard root dir, use
+       If you want to use a non-standard root dir in the current session, use
             GCRCatalogs.set_root_dir('/path/to/your/root_dir')
+
+       To also make that value the default in future sessions, use
+            GCRCatalogs.set_root_dir('/path/to/your/root_dir', write_to_config=True)
+
     """
 
-    def __init__(self, site_config_path=None):
+    def __init__(self, site_config_path=None, user_config_path=None):
         self._site_config_path = site_config_path
+        self._user_config_path = user_config_path
         self._default_root_dir = None
         self._custom_root_dir = None
         self._site_config = {}
         self._site_info = self._get_site_info()
 
+        # User config has highest priority
+        if self._user_config_path:
+            if os.path.exists(_USER_CONFIG_PATH):
+                d = load_yaml_local(_USER_CONFIG_PATH)
+                if 'root_dir' in d.keys():
+                    self._default_root_dir = d['root_dir']
+                    return
```

Comment:
```suggestion
        if self._user_config_path and os.path.isfile(self._user_config_path):
            d = load_yaml_local(_USER_CONFIG_PATH)
            if 'root_dir' in d:
                self._default_root_dir = d['root_dir']
                return
```

Reply:
I'd like to keep the nested `if` but I'll incorporate the other suggestions in my next push.

Reply:
OK. BTW, the `_USER_CONFIG_PATH` within the call `load_yaml_local` should probably be `self._user_config_path` as well? 

But I think this would be easier once we implement the user config class.

Reply:
Yes, I noticed that and changed it for the user config class.